pkg/gitops/flux/cluster.go

```diff
@@ -0,0 +1,261 @@
+package flux
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/git"
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/providers"
+	"github.com/aws/eks-anywhere/pkg/validations"
+)
+
+// fluxForCluster bundles the Flux struct with a specific clusterSpec, so that all the git and file write
+// operations for the clusterSpec can be done in each structure method.
+type fluxForCluster struct {
+	*Flux
+	clusterSpec      *cluster.Spec
+	datacenterConfig providers.DatacenterConfig
+	machineConfigs   []providers.MachineConfig
+}
+
+func newFluxForCluster(flux *Flux, clusterSpec *cluster.Spec, datacenterConfig providers.DatacenterConfig, machineConfigs []providers.MachineConfig) *fluxForCluster {
+	return &fluxForCluster{
+		Flux:             flux,
+		clusterSpec:      clusterSpec,
+		datacenterConfig: datacenterConfig,
+		machineConfigs:   machineConfigs,
+	}
+}
+
+// commitFluxAndClusterConfigToGit commits the cluster configuration file to the flux-managed git repository.
+// If the remote repository does not exist it will initialize a local repository and push it to the configured remote.
+// It will generate the kustomization file and marshal the cluster configuration file to the required locations in the repo.
+// These will later be used by Flux and our controllers to reconcile the repository contents and the cluster configuration.
+func (fc *fluxForCluster) commitFluxAndClusterConfigToGit(ctx context.Context) error {
+	logger.Info("Adding cluster configuration files to Git")
+	config := fc.clusterSpec.FluxConfig
+
+	if err := fc.validateLocalConfigPathDoesNotExist(); err != nil {
+		return fmt.Errorf("validating local config path not exists: %v", err)
```

Comment:
nit: "local config path does not exist" ?

Reply:
this actually we need "validating local config path not exists"

Reply:
the method has proper error info. gonna remove the error statement here